### PR TITLE
Adjust to new size analysis implementation.

### PR DESCRIPTION
**Pull request by yifanyang**
The metrics backend for size analysis is revamped. We should be able to see some UI improvements ([a demo report](https://github.com/yifanyang/javascript/pull/3#issuecomment-760919595)) in the size analysis report.

The new implementation returns 200 (OK) instead of 202 (ACCEPTED). Therefor the script should also treat 200 as success.

### CODE REVIEW

**Comment by yifanyang** [comment]
@Feiyang1 The original size report are missing. It looks like the that job ([Health Metrics / Binary Size (pull_request)](https://github.com/firebase/firebase-js-sdk/pull/4300/checks?check_run_id=1710344549)) actually failed with:
```
(node:4844) UnhandledPromiseRejectionWarning: Error: Could not resolve entry module (packages/storage/exp/dist/index.browser.cjs.js).
    at error (/home/runner/work/firebase-js-sdk/firebase-js-sdk/node_modules/rollup/dist/shared/rollup.js:5265:30)
    at ModuleLoader.loadEntryModule (/home/runner/work/firebase-js-sdk/firebase-js-sdk/node_modules/rollup/dist/shared/rollup.js:18454:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at Object.main (/home/runner/work/firebase-js-sdk/firebase-js-sdk/node_modules/ts-node/src/bin.ts:198:14)
    at Object.<anonymous> (/home/runner/work/firebase-js-sdk/firebase-js-sdk/node_modules/ts-node/src/bin-script.ts:5:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
```
But it is still marked as success on CI. I recalled that you once had some fix for a similar issue. Is that in?

**Comment by Feiyang1** [comment]
Yes, https://github.com/firebase/firebase-js-sdk/pull/4010. Is it a case this PR doesn't cover? @yifanyang

File: scripts/size_report/size_report_helper.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export function upload(
     console.log(`Response status code: ${response.statusCode}`);
     response.on('data', console.log);
     response.on('end', () => {
-      if (response.statusCode !== 202) {
+      if (response.statusCode !== 200 && response.statusCode !== 202) {
```
Comment (Feiyang1) [inline scripts/size_report/size_report_helper.ts:80]: Should we still accept 202?

**Review by Feiyang1 — APPROVED** [review]

File: scripts/size_report/size_report_helper.ts (hunk 1 above)
Comment (yifanyang) [reply scripts/size_report/size_report_helper.ts:80]: Yes, for now. The api for the old size report still returns 202. I'm planning to migrate it to 200 as well. I'll remove the 202 here (and a few other places in android repo) when that happens.

**Comment by yifanyang** [comment]
> Yes, #4010. Is it a case this PR doesn't cover? @yifanyang

Got it. We can probably investigate on that unhandled reject later. I'm merging this one now to unblock other developers.

**Pull request merged**